graph-gateway/src/network/indexer_indexing_progress_resolver.rs

```diff
@@ -223,3 +238,49 @@ impl IndexingProgressResolver {
         self.resolve_with_cache(url, indexer_deployments).await
     }
 }
+
+// TODO: Change visibility once integration tests are moved
+pub async fn send_requests(
+    client: &reqwest::Client,
+    status_url: Url,
+    indexings: &[DeploymentId],
+    batch_size: usize,
+) -> HashMap<DeploymentId, Result<Vec<ChainStatus>, IndexingProgressFetchError>> {
+    debug_assert_ne!(batch_size, 0, "batch size must be greater than 0");
+
+    let responses = {
+        let requests = indexings.chunks(batch_size).map(|deployments| {
+            let status_url = status_url.clone();
+            async move {
+                let result =
+                    indexers::indexing_progress::send_request(client, status_url, deployments)
+                        .await;
+
+                let result = match result {
+                    Err(err) => {
+                        // Mark all indexings' as failed
+                        return deployments
+                            .iter()
+                            .map(|deployment_id| (*deployment_id, Err(err.clone())))
+                            .collect::<HashMap<_, _>>();
+                    }
+                    Ok(res) => res,
+                };
```

Comment:
I don't think we should be marking all the indexings as failed if the request fails intermittently.

Reply:
Note that the `async` block is within a `map(|deployments|{...})` function. 

Here, if a request fails, we mark all the deployment IDs in that batch with the returned error. 

Let me add some comments so it is clearer :)